Clean up rtos::Thread deprecation warnings

kjbracey opened the pull request:
### Description

Static rtos::Thread methods and signal methods have been deprecated. Remove
all references in the main code, and most of the tests. Some tests of
the deprecated APIs themselves remain.

### Pull request type

    [ ] Fix
    [x] Refactor
    [ ] Target update
    [ ] Functionality change
    [ ] Breaking change

This isn't a bug fix, just avoids the build warnings, so probably not for a patch release. Could be rc2 perhaps.

kjbracey commented on TESTS/mbed_drivers/mem_trace/main.cpp:
```diff
@@ -391,7 +391,7 @@ static void test_case_multithread_malloc_free()
     }
 
-    Thread::wait(wait_time_us);
+    ThisThread::sleep_for(wait_time_us);
```
I note in passing that this takes milliseconds not microseconds. I wonder if this is supposed to run for 10 seconds or 10 milliseconds? Not changing it in this PR, anyway.

bmcdonnell-ionx commented:
I added the additional substitutions I found in #8127 to a branch in my fork, https://github.com/bmcdonnell-ionx/mbed-os/tree/thread_deprecations.

@kjbracey-arm, have a look and see if you want to pull any of my (3) commits there into this.

kjbracey commented:
Thanks!

For the tests, the return values of the signal functions are different, so your transformation doesn't work directly - would fail to compile.

I transformed some of the tests to test the new API, but decided it was simplest to leave that signal one testing the original. Maybe revisit and transform later.

Will incorporate the other 2 patches.

0xc0170 commented:
@ARMmbed/mbed-os-core Needs your review now

cmonr commented:
/morph build

0xc0170 commented:
The failures related to the changes, please review

kjbracey committed:
Clean up Thread deprecation warnings

Static Thread methods and signal methods have been deprecated. Remove
all references in the main code, and most of the tests. Some tests of
the deprecated APIs themselves remain.

kjbracey commented:
Rebased - couple more conversions needed in features/cellular/TESTS/api/cellular_network and  TEST_APPS/device/nanostack_mac_tester

cmonr commented:
@kjbracey-arm It looks like this still needs to pass unittests before it can go thorugh CI.

kjbracey commented:
Unit tests fixed

NirSonnenschein commented:
/morph build
